scala/lambdas/custodial-copy-queue-creator/src/main/scala/uk/gov/nationalarchives/custodialcopyqueuecreator/Lambda.scala

```diff
@@ -0,0 +1,85 @@
+package uk.gov.nationalarchives.custodialcopyqueuecreator
+
+import cats.effect.*
+import cats.syntax.all.*
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import io.circe.{Decoder, Encoder, HCursor, Json, JsonObject}
+import pureconfig.ConfigReader
+import pureconfig.generic.derivation.default.*
+import sttp.capabilities.fs2.Fs2Streams
+import uk.gov.nationalarchives.dp.client.EntityClient
+import uk.gov.nationalarchives.DASQSClient
+import uk.gov.nationalarchives.custodialcopyqueuecreator.Lambda.*
+import uk.gov.nationalarchives.custodialcopyqueuecreator.Lambda.MessageBody.*
+import uk.gov.nationalarchives.utils.LambdaRunner
+import uk.gov.nationalarchives.dp.client.fs2.Fs2Client
+import uk.gov.nationalarchives.utils.EventDecoders.given
+import io.circe.parser.decode
+import uk.gov.nationalarchives.DASQSClient.FifoQueueConfiguration
+import uk.gov.nationalarchives.dp.client.EntityClient.EntityType
+
+import scala.jdk.CollectionConverters.*
+import java.util.UUID
+
+class Lambda extends LambdaRunner[SQSEvent, Unit, Config, Dependencies]:
+
+  override def handler: (SQSEvent, Config, Dependencies) => IO[Unit] = { (sqsEvent, config, dependencies) =>
+    sqsEvent.getRecords.asScala.toList.parTraverse { record =>
+      for {
+        _ <- IO.println(s"AAAAAAAAAAAAAAAAAAA ${record.getReceiptHandle}")
+        messageBody <- IO.fromEither(decode[MessageBody](record.getBody))
```

Comment:
They're not the same. `CompactEntity` has the id as `String` because it's the `io:<uuid>` format. `MessageBody` has parsed that so the ID is the UUID and the type is `IoMessageBody`, `CoMessageBody` etc.